### ISSUE TITLE
continuous-integration/travis-ci/pr always fail in x86

### ISSUE THREAD

**Issue by teawater** [kata-containers/runtime#1492]
```
	firecracker hypervisor path (FCPATH) : /usr/bin/firecracker
	assets path (PKGDATADIR) : /usr/share/kata-containers
	proxy+shim path (PKGLIBEXECDIR) : /usr/libexec/kata-containers
     BUILD   /home/travis/gopath/src/github.com/kata-containers/runtime/kata-runtime
     CONFIG  cli/config/configuration-qemu.toml
     CONFIG  cli/config/configuration-fc.toml
     BUILD   /home/travis/gopath/src/github.com/kata-containers/runtime/containerd-shim-kata-v2
# github.com/kata-containers/runtime/cli/containerd-shim-kata-v2
/home/travis/.gimme/versions/go1.11.7.linux.amd64/pkg/tool/linux_amd64/link: running gcc failed: exit status 1
/usr/bin/ld: /tmp/go-link-503628299/000010.o: unrecognized relocation (0x2a) in section `.text'
/usr/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
make: *** [/home/travis/gopath/src/github.com/kata-containers/runtime/containerd-shim-kata-v2] Error 2
The command "make" failed and exited with 2 during .
Your build has been stopped.
```

**Comment by teawater** [comment]
https://github.com/kata-containers/runtime/pull/1491

**Comment by chavafg** [comment]
for reference: https://github.com/golang/go/issues/31293

**Comment by teawater** [comment]
> for reference: [golang/go#31293](https://github.com/golang/go/issues/31293)

@lifupan could you take a look on this?

**Comment by teawater** [comment]
This is what I report to travis and their reply:
```
Hui Zhu

Apr 8, 22:09 EDT

Always get some fails like:
https://travis-ci.org/kata-containers/runtime/jobs/517090263
# github.com/kata-containers/runtime/cli/containerd-shim-kata-v2
/home/travis/.gimme/versions/go1.11.8.linux.amd64/pkg/tool/linux_amd64/link: running gcc failed: exit status 1
/usr/bin/ld: /tmp/go-link-891013085/000010.o: unrecognized relocation (0x2a) in section `.text'
/usr/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
make: *** [/home/travis/gopath/src/github.com/kata-containers/runtime/containerd-shim-kata-v2] Error 2
The command "make" failed and exited with 2 during .

–––––––––––––––
The following details will be sent along to help us help you better, but you may also edit them as you like.

Page: https://travis-ci.org/
```
```
Hey there,

Thanks so much for reaching out!

This is just a quick note to let you know we've received your message. Our support hours are typically between 10:00 AM CET - 10:00 PM CET, Monday to Friday. We will do our very best to respond as soon as possible.

If you have any additional information for us, like screenshots, build URLs, or things you've tried, please reply directly to this email.

Cheers,

The Travis CI Builders
```

**Comment by teawater** [comment]
https://support.travis-ci.com/hc/en-us/requests/6220

**Comment by grahamwhaley** [comment]
Right, basic issue is (was) that the new go was built with a newer gcc, and Travis (by default, for us), runs an old Ubuntu (14), which then has this mismatch.
I see @chavafg has PR'd the updates of the base ubuntu version to the travis files - now, iirc, last time we discussed/tried that (maybe 2 years ago), there was a reason we did not move - it might have been that the free Travis did not really support it at the time. I'll have a look at those update PRs soon...
Anyway, not our bug, but issue triggered by our updating golang version and using a rather old Ubuntu version on Travis. Now we race to see if we fix this on our travis before the upstream golang fixes it with a rebuild..
